**Автор: Анна Смелова**

## Решение домашнего задания к уроку 3 “Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных”

In [1]:
import numpy as np
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Задача
#### Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

**<font color='darkblue'>Решение:</font>**<br>

In [3]:
x = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
n = len(x)
print(f'n={n}')

n=20


In [4]:
m = 0
for i in x:
    m += i
m = m / n
print(f'Среднее арифметическое:{m}')

Среднее арифметическое:65.3


In [5]:
s = 0
for i in x:
    s += (i - m)**2
s = (s / n)**(1 / 2)
print(f'Среднее квадратичное отклонение:{s}')

Среднее квадратичное отклонение:30.823854398825596


In [6]:
d = 0
for i in x:
    d += (i - m)**2
d1 = d / n
d2 = d / (n - 1)
print(f'Смещенная оценка дисперсии: {d1}')
print(f'Несмещенная оценка дисперсии: {d2}')

Смещенная оценка дисперсии: 950.11
Несмещенная оценка дисперсии: 1000.1157894736842


### 2. Задача
#### В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

**<font color='darkblue'>Решение:</font>**<br>
Ситуация 1: 2 белых из 1 ящика и 1 белый из 2 ящика: $P(2-1)=\frac{C_5^2}{C_8^2}\cdot\frac{C_5^1\cdot C_7^3}{C_{12}^4}$<br>
Ситуация 2: 1 белый из 1 ящика и 2 белых из 2 ящика: $P(1-2)=\frac{C_5^1\cdot C_3^1}{C_8^2}\cdot\frac{C_5^2\cdot C_7^2}{C_{12}^4}$<br>
Ситуация 3: 0 белых из 1 ящика и 3 белых из 2 ящика: $P(0-3)=\frac{C_3^2}{C_8^2}\cdot\frac{C_5^3\cdot C_7^1}{C_{12}^4}$<br>

In [7]:
p21 = (combinations(5, 2) * combinations(5, 1) * combinations(7, 3)) / (combinations(8, 2) * combinations(12, 4))
p12 = (combinations(5, 1) * combinations(3, 1) * combinations(5, 2) * combinations(7, 2)) / (combinations(8, 2) * combinations(12, 4))
p03 = (combinations(3, 2) * combinations(5, 3) * combinations(7, 1)) / (combinations(8, 2) * combinations(12, 4))
print(f'p21={p21}')
print(f'p12={p12}')
print(f'p03={p03}')

p21=0.12626262626262627
p12=0.22727272727272727
p03=0.015151515151515152


$P(3white)=P(2-1)+P(1-2)+P(0-3)$

In [8]:
p = p21 + p12 + p03
print(f'P={p}')

P=0.3686868686868687


**<font color='darkblue'>Ответ: вероятность того, что 3 мяча белые $0.3686868686868687$.</font>**

### 3. Задача
#### На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

**<font color='darkblue'>Решение:</font>**<br>
Событие $A$-попали в мишень, событие $B_1$-попал первый спортсмен, $B_2$-попал второй спортсмен, $B_3$-попал третий спортсмен.<br>
Априорная вероятность: $P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$<br>
$P(A|B_1)=0.9$<br>
$P(A|B_2)=0.8$<br>
$P(A|B_3)=0.6$<br>
Формула полной вероятности: $P(A)=P(B_1)\cdot P(A|B_1)+P(B_2)\cdot P(A|B_2)+P(B_3)\cdot P(A|B_3)=\frac{1}{3}\cdot0.9+\frac{1}{3}\cdot0.8+\frac{1}{3}\cdot0.6=\frac{23}{30}$<br><br>
Апостериорные вероятности:<br>
выстрел произведен первым спортсменом: $P(B_1|A)=\frac{P(B_1)\cdot P(A|B_1)}{P(A)}=\frac{\frac{1}{3}\cdot0.9}{\frac{23}{30}}=\frac{30\cdot9}{3\cdot10\cdot23}=\frac{9}{23}=0.391304347826087$<br>
выстрел произведен вторым спортсменом: $P(B_2|A)=\frac{P(B_2)\cdot P(A|B_2)}{P(A)}=\frac{\frac{1}{3}\cdot0.8}{\frac{23}{30}}=\frac{30\cdot8}{3\cdot10\cdot23}=\frac{8}{23}=0.34782608695652173$<br>
выстрел произведен третьим спортсменом: $P(B_3|A)=\frac{P(B_3)\cdot P(A|B_3)}{P(A)}=\frac{\frac{1}{3}\cdot0.6}{\frac{23}{30}}=\frac{30\cdot6}{3\cdot10\cdot23}=\frac{6}{23}=0.2608695652173913$

In [9]:
print(f'P(B_1|A)={9 / 23}')
print(f'P(B_2|A)={8 / 23}')
print(f'P(B_3|A)={6 / 23}')

P(B_1|A)=0.391304347826087
P(B_2|A)=0.34782608695652173
P(B_3|A)=0.2608695652173913


**<font color='darkblue'>Ответ: вероятность того, что выстрел произведен первым спортсменом $0.391304347826087$,<br>
выстрел произведен вторым спортсменом $0.34782608695652173$,<br>
выстрел произведен третьим спортсменом $0.2608695652173913$.</font>**

### 4. Задача
#### В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

**<font color='darkblue'>Решение:</font>**<br>
Количество студентов $n_A=n_B=n$, $n_C=2n$<br>
Событие $S$-сдал сессию, событие $A$-студент факультета A, $B$-студент факультета B, $C$-студент факультета C.<br>
Априорная вероятность: $P(A)=P(B)=\frac{n}{n+n+2n}=\frac{1}{4}$, $P(C)=\frac{2n}{n+n+2n}=\frac{1}{2}$<br>
$P(S|A)=0.8$<br>
$P(S|B)=0.7$<br>
$P(S|C)=0.9$<br>
Формула полной вероятности: $P(S)=P(A)\cdot P(S|A)+P(B)\cdot P(S|B)+P(C)\cdot P(S|C)=\frac{1}{4}\cdot0.8+\frac{1}{4}\cdot0.7+\frac{1}{2}\cdot0.9=\frac{8+7+18}{4\cdot10}=\frac{33}{40}$<br><br>
Апостериорные вероятности:<br>
сдал студент с факультета A: $P(A|S)=\frac{P(A)\cdot P(S|A)}{P(S)}=\frac{\frac{1}{4}\cdot0.8}{\frac{33}{40}}=\frac{40\cdot8}{4\cdot10\cdot33}=\frac{8}{33}=0.24242424242424243$<br>
сдал студент с факультета B: $P(B|S)=\frac{P(B)\cdot P(S|B)}{P(S)}=\frac{\frac{1}{4}\cdot0.7}{\frac{33}{40}}=\frac{40\cdot7}{4\cdot10\cdot33}=\frac{7}{33}=0.21212121212121213$<br>
сдал студент с факультета C: $P(C|S)=\frac{P(C)\cdot P(S|C)}{P(S)}=\frac{\frac{1}{2}\cdot0.9}{\frac{33}{40}}=\frac{40\cdot9}{2\cdot10\cdot33}=\frac{18}{33}=0.5454545454545454$

In [10]:
print(f'P(A|S)={8 / 33}')
print(f'P(B|S)={7 / 33}')
print(f'P(C|S)={18 / 33}')

P(A|S)=0.24242424242424243
P(B|S)=0.21212121212121213
P(C|S)=0.5454545454545454


**<font color='darkblue'>Ответ: вероятность того, что сдал студент с факультета A $0.(24)$,<br>
вероятность того, что сдал студент с факультета B $0.(21)$,<br>
вероятность того, что сдал студент с факультета C $0.(54)$.</font>**

### 5. Задача
#### Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

**<font color='darkblue'>Решение:</font>**<br>
Событие $B_1$-вышла из строя первая деталь, $P(B_1)=0.1$<br>
Событие $B_2$-вышла из строя вторая деталь, $P(B_2)=0.2$<br>
Событие $B_3$-вышла из строя третья деталь, $P(B_3)=0.25$<br><br>
а) Вышли из строя все детали: $P(B_1B_2B_3)=P(B_1)\cdot P(B_2)\cdot P(B_3)=0.1\cdot0.2\cdot0.25=0.005$.<br><br>
б) Вышли из строя только две детали: $P(=2)=P(B_1B_2\overline{B_3})+P(B_1\overline{B_2}B_3)+P(\overline{B_1}B_2B_3)=P(B_1)\cdot P(B_2)\cdot (1-P(B_3))+P(B_1)\cdot (1-P(B_2))\cdot P(B_3)+(1-P(B_1))\cdot P(B_2)\cdot P(B_3)=0.1\cdot0.2\cdot0.75+0.1\cdot0.8\cdot0.25+0.9\cdot0.2\cdot0.25=0.015+0.02+0.045=0.08$.<br><br>
в) Вышла из строя хотя бы одна деталь: $P(\geq1)=1-P(\overline{B_1B_2B_3})=(1-P(B_1))\cdot(1-P(B_2))\cdot(1-P(B_3))=0.9\cdot0.8\cdot0.75=0.46$<br><br>
г) Вышли из строя от одной до двух деталей: $P(1\geq\leq2)=P(B_1\overline{B_2B_3})+P(\overline{B_1}B_2\overline{B_3})+P(\overline{B_1B_2}B_3)+P(=2)=P(B_1)\cdot(1-P(B_2))\cdot(1-P(B_3))+(1-P(B_1))\cdot P(B_2)\cdot(1-P(B_3))+(1-P(B_1))\cdot(1-P(B_2))\cdot P(B_3)+P(=2)=0.1\cdot0.8\cdot0.75+0.9\cdot0.2\cdot0.75+0.9\cdot0.8\cdot0.25+0.08=0.06+0.135+0.18+0.08=0.455.$

In [11]:
print(f'P(B1B2B3)={0.1*0.2*0.25}')
print(f'P(=2)={0.1*0.2*0.75+0.1*0.8*0.25+0.9*0.2*0.25}')
print(f'P(>=1)={1-0.9*0.8*0.75}')
print(f'P(1>==<2)={0.1*0.8*0.75+0.9*0.2*0.75+0.9*0.8*0.25+0.08}')

P(B1B2B3)=0.005000000000000001
P(=2)=0.08000000000000002
P(>=1)=0.45999999999999996
P(1>==<2)=0.455


**<font color='darkblue'>Ответ: a) $0.005$,<br> б) $0.08$,<br> в) $0.46$,<br> г) $0.455$.</font>**